### ISSUE TITLE
Headercolumn translation within data-title

### ISSUE THREAD

**Issue by Disane87** [awerlang/angular-responsive-tables#15]
Currently I'm using "ngTasty" to get a proper table (server side) of my data. Now I need to translate the header columns for mobile devices. For the reponsive part, I use your angular-responsive.

The translation within the "normal" header cols works:

``` html
<thead>
                        <th class="mobileview"  ng-repeat="h in header.columns" > {{ h.name | uppercase | translate:translationData }}</th>
                      </thead>
```

Nothing works, to get the translated column names for smaler devices, i.e.:

``` html
<td data-title="'NAME' | translate">
```

or

``` html
<td data-title="translate['PREIS']">{{ row.Preis | currency }}</td>
```

My entire table looks like this:

``` html
<div tasty-table bind-filters="filters" bind-resource-callback="getResource" bind-init="init">
                    <table wt-responsive-table class="table table-striped table-condensed">
                      <thead>
                        <th class="mobileview"  ng-repeat="h in header.columns" > {{ h.name | uppercase | translate:translationData }}</th>
                      </thead>
                      <!-- <th class="mobileview" ng-repeat="h in header.columns" style="display:none;"> {{h.name}} </th> -->
                      <tbody>
                        <tr ng-repeat="row in rows">
                          <!-- <td>{{ row.ID }}</td> -->
                          <td>
                            <span action-button color="btn-success" itemid="row.ID" type="'recipe'"></span>
                          </td>
                          <td data-title="'NAME' | translate">{{ row.Name }}</td>
                          <td data-title="DosierungMin">{{row.DosierungMin}}%</td>
                          <td data-title="DosierungMax">{{row.DosierungMax}}%</td>
                          <td data-title="Dosierung">
                            <!-- <p ng-if="flavor.DosierungMax > 0"> -->
                              <span ng-if="row.DosierungMax == 0" class="animate-if label label-success">
                                <i class="fa fa-times fa-fw" aria-hidden="true"></i>
                              </span>
                              <span ng-if="row.DosierungMax < 6 && row.DosierungMax > 0" class="animate-if label label-primary">
                                <i class="fa fa-arrow-down fa-fw" aria-hidden="true"></i>
                              </span>
                              <span ng-if="row.DosierungMax >= 6 && row.DosierungMax < 10" class="animate-if label label-warning">
                                <i class="fa fa-arrows-h fa-fw" aria-hidden="true"></i>
                              </span>
                              <span ng-if="row.DosierungMax >= 10" class="animate-if label label-danger">
                                <i class="fa fa-arrow-up fa-fw" aria-hidden="true"></i>
                              </span>
                            <!-- </p> -->
                          </td>
                          <!-- <td>{{ row.DosierungMin  | volumne }}</td> -->
                          <!-- <td>{{ row.DosierungMax }}mg</td> -->
                          <td data-title="Reifezeit">{{ row.Reifezeit }} {{ 'DAYS' | translate:translationData }}</td>
                          <td data-title="Menge">{{ row.Menge  | volumne }}</td>

                          <td data-title="translate['PREIS']">{{ row.Preis | currency }}</td>
                          <td data-title="Preis100ml">{{ row.Preis100ml | currency }}</td>
                          <td data-title="Hersteller">{{ row.Hersteller }}</td>
                          <!-- <td>{{ row.Herstellungsdatum | amUtc | amDateFormat:'L' }}</td> -->

                          <!-- <td>{{ row.Bewertung }}</td> -->
                          <td data-title="Eigenschaften" ng-bind-html="row.Eigenschaften"></td>
                        </tr>
                      </tbody>
                    </table>
                    <div tasty-pagination bind-list-items-per-page="listItemsPerPage"></div>
                  </div>
```

Screenshots:
1) Working "unresponsive" on larger screens:
<img width="1209" alt="bildschirmfoto 2016-08-02 um 09 20 53" src="https://cloud.githubusercontent.com/assets/956158/17320197/8f4b85a4-5892-11e6-8ff1-fa75f894fdb8.png">

2) Not working on mobile:
<img width="460" alt="bildschirmfoto 2016-08-02 um 09 21 00" src="https://cloud.githubusercontent.com/assets/956158/17320203/941bbd10-5892-11e6-9e52-44162d128f0e.png">

Do you have a clue?

**Comment by awerlang** [comment]
Hello

It should work if you use interpolation on the `data-title` attribute, e.g. `data-title="{{'NAME' | translate}}"`.
Yet, the attribute `data-title` is not part of the public API, so it shouldn't be used explicitly on the markup. Have you tried removing `data-title`? This way the textual content of the `<th>` would be used as headers on the rows. It is expected to work even with dynamicly generated headers like you have, but a similar issue was already reported at #12.

**Comment by Disane87** [comment]
Thank you for your fast reply. I've already tried it without data-title, but without it, no title will be shown and I don't know why. With interpolation it works like a charm, but is only workaround, as I discovered.

**Comment by Disane87** [comment]
Anything new to this? Do you need a plunkr or something else to investigate this? Please let me know, if you need something :-)

**Comment by awerlang** [comment]
I have an idea of what's going on, looks like the columns, interpolation etc are linked after this component runs, that's why it works only with static text on `<th>`. It is odd to me because I use dynamic headers on my projects, and it works. Must be some other thing I'm doing there. I'll let you know if I get a more conclusive finding. Using this plunker to verify: https://plnkr.co/edit/eEmELoJot3pveK6CGFLG?p=preview

**Comment by Disane87** [comment]
Ok great, thank you! :)

**Comment by awerlang** [comment]
Closed as duplicate of #12.

**Comment by awerlang** [comment]
released in v0.2.0. plunker updated: https://embed.plnkr.co/eEmELoJot3pveK6CGFLG/

**Comment by Disane87** [comment]
Thanks for the fix. Unfortunatly it will work without tastyTable, not in combination. I've prepared a plunker to demonstrate my approach: http://plnkr.co/edit/VO4XwXfn4zZxJuIkYPlD

**Comment by awerlang** [comment]
It shows up blank here, is that right? I noticed the json request was returning HTTP 400, but even changing `$http.post()` to `$http.get()`, the problem persists. I removed `wt-responsive-table` from your table to no avail.

I'm thinking of going through the implementation once more, if we have this as a repro it would be great.

**Comment by Disane87** [comment]
I've tested it again in another browser and It starts in edit-mode on plunker, which works:
![bildschirmfoto 2016-09-01 um 22 02 30](https://cloud.githubusercontent.com/assets/956158/18182325/cc47bbf2-708f-11e6-8b46-702c69987469.png)

I guess, if you fork it, it will work. I don't know why the embedded mode actually is not working.

Edit:
it was my fault, I actually didn't freeze my code. 😞 
Here is the tested link: http://embed.plnkr.co/VO4XwXfn4zZxJuIkYPlD/

**Comment by awerlang** [comment]
Good! I'll take a look.

**Comment by awerlang** [comment]
Issue was resolved with v0.3.0

http://embed.plnkr.co/axfjfm47jv6BWzIi4bT6/

**Comment by Disane87** [comment]
Now it works, tanks you! :)
